mlir/lib/Dialect/Linalg/IR/CMakeLists.txt

```
@@ -13,6 +13,7 @@ add_mlir_dialect_library(MLIRLinalgDialect
   MLIRLinalgOpsEnumsIncGen
   MLIRLinalgOpsIncGen
   MLIRLinalgStructuredOpsIncGen
+  MLIRShardingInterface
```

```suggestion
  MLIRShardingInterfaceIncGen
```

I think you're only trying to solve the dependency on the generated files, so this should do it.